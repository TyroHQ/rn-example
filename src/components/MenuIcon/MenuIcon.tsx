import React from "react";
import { Text, View } from "react-native";
import styled from "styled-components";
import { Placeholder } from "../Placeholder";

const MenuIconContainer = styled(View)`
  background-color: lightgrey;
  padding: 10px;
`;

export class MenuIcon extends React.Component<{}> {
  static defaultProps: MenuIcon["props"];
  render() {
    return <Placeholder>MenuIcon</Placeholder>;
  }
}
