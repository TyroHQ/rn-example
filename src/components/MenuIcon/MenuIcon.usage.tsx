import React from "react";
import { boolean, text, number } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";
import { MenuIcon } from "./";

export const basic = () => <MenuIcon />;
