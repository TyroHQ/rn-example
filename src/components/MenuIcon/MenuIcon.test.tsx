import { shallow } from "enzyme";
import * as usage from "./MenuIcon.usage";

describe("<MenuIcon />", () => {
  for (const example in usage) {
    it(`renders well in ${example} mode`, () => {
      const x = (usage as any)[example]();
      const wrapper = shallow(x);
      expect(wrapper).toMatchSnapshot();
    });
  }
});
