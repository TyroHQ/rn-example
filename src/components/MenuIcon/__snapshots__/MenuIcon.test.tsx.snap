// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuIcon /> renders well in basic mode 1`] = `
<Placeholder
  orientation="row"
>
  MenuIcon
</Placeholder>
`;
